src/trie.js

```diff
@@ -82,30 +82,31 @@ var marexandre;
     Trie.prototype.getIndices = function(_text_) {
       var self = this;
       var result = [];
-      var copy = '';
-      var tmpTrie = self.list;
+      var remainingText = '';
+      var currentNode = self.list;
       var start = -1, end = -1;
 
       for (var i = 0, imax = _text_.length; i < imax; i++) {
-        copy = _text_.slice(i);
+        remainingText = _text_.slice(i);
 
         // TODO: Need to refactor this loop :(
-        for (var j = 0, jmax = copy.length; j < jmax; j++) {
-          var c = copy[j];
-          var exists = tmpTrie.children.hasOwnProperty(c.toString());
+        for (var j = 0, jmax = remainingText.length; j < jmax; j++) {
+          var c = remainingText[j];
 
-          if (exists) {
-            tmpTrie = tmpTrie.children[c];
+          if (self.hasChildrenWithValue(currentNode, c)) {
```

Comment:
it sounds bit big refactoring. i need to create another javascript object and change many part of codes in trie.js file and also unit tests. I don't think we should do this in this small bug fix issue. I'll make another chore ticket for this if you need

```
  var Node = (function() {
    function Node(children, value, is_end) {
      this.children = children;
      this.value = value;
      this.is_end = is_end;
    }

    /**
     * hasChildrenWithValue returns if node of trie has a child
     * @param  {trie} node Object tire node
     * @param  {String} char value of trie node
     * @type {boolean} true if node has a child
     */
    Node.prototype.hasChildrenWithValue = function(char) {
      return this.children.hasOwnProperty(char.toString());
    };

    return Node;
  })();
```